### PR TITLE
feat: add https middleware package

### PR DESCRIPTION
**Pull request by edgarmueller**

### CODE REVIEW

File: packages/nestjs-https-redirect/.gitignore [hunk 1]
```diff
@@ -0,0 +1,156 @@
+### https://raw.github.com/github/gitignore/d2c1bb2b9c72ead618c9f6a48280ebc7a8e0dff6/Node.gitignore
```
Comment (ottovw) [inline packages/nestjs-https-redirect/.gitignore:1]: .DS_Store is missing

File: packages/nestjs-https-redirect/package.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    "@types/jest": "^24.9.1",
+    "husky": "^3.0.9",
+    "jest": "^26.6.3",
+    "semantic-release": "^17.3.1",
+    "ts-jest": "^26.4.4",
+    "tsdx": "^0.14.1",
+    "tslib": "^1.10.0",
+    "typescript": "^3.7.2"
+  },
+  "dependencies": {
```
Comment (ottovw) [inline packages/nestjs-https-redirect/package.json:77]: review dependencies

File: packages/nestjs-https-redirect/src/https-redirect.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  use(req: any, res: any, next: () => void) {
+    const enabled = this.options.enabled;
+
+    if (!enabled) {
+      next();
+      return;
+    }
+
+    const isHttpProto = req.header('x-forwarded-proto') === 'http';
```
Comment (ottovw) [inline packages/nestjs-https-redirect/src/https-redirect.middleware.ts:18]: this is not needed.

File: packages/nestjs-https-redirect/src/https-redirect.middleware.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    if (!req.secure && isHttpProto) {
+      const redirectUrl = `https://${req.hostname}${req.originalUrl}`;
```
Comment (ottovw) [inline packages/nestjs-https-redirect/src/https-redirect.middleware.ts:21]: are query parameters redirected as well?

File: packages/nestjs-https-redirect/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Kittgen Nestjs Https Redirect
+
+HTTPS redirect middleware for NestJS.
+
+[Check the main page for Kittgen](../../README.md) for further information.
+
```
Comment (ottovw) [inline packages/nestjs-https-redirect/README.md:6]: use docu template from authorization

**Review by ottovw — CHANGES_REQUESTED** [review]

File: packages/nestjs-https-redirect/src/https-redirect.middleware.ts (hunk 4 above)
Comment (edgarmueller) [reply packages/nestjs-https-redirect/src/https-redirect.middleware.ts:21]: AFAIK, yes, see https://expressjs.com/de/api.html `req.originalUrl`

File: packages/nestjs-https-redirect/README.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+### Installation
+
+```bash
+npm i @kittgen/nestjs-https-redirect
+```
+
+### Register your middleware
+
+First, make sure enable `trust proxy` for you app:
```
Comment (ottovw) [inline packages/nestjs-https-redirect/README.md:17]: only if behind reverse proxy (e.g. Heroku or Ngnix)

File: packages/nestjs-https-redirect/README.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+app.enable('trust proxy`);
+```
+
+Then in your application module, set up the middleware:
+
+```ts
+export class ApplicationModule {
+    configure(consumer: MiddlewareConsumer) {
+        consumer
+          .apply(HttpsRedirectMiddleware)
```
Comment (ottovw) [inline packages/nestjs-https-redirect/README.md:29]: how to pass in configuration?

**Review by ottovw — CHANGES_REQUESTED** [review]

**Comment by ottovw** [comment]
insert reference into main README.md

File: packages/nestjs-https-redirect/README.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
+# Kittgen Nestjs HTTPS redirect middleware
```
Comment (ottovw) [inline packages/nestjs-https-redirect/README.md:1]: add reference to root-level README.md

**Review by ottovw — CHANGES_REQUESTED** [review]

**Review by ottovw — APPROVED** [review]
👍 💯 🚀

**Pull request merged**